### PR TITLE
Move some common Scalatra code into socrata-thirdparty-utils

### PR DESCRIPTION
**Pull request by urmilan**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -17,6 +17,7 @@ val commonDeps = Seq(
   "com.rojoma"        %% "simple-arm"          % "[1.2.0,2.0.0)",
   "com.rojoma"        %% "simple-arm-v2"       % "2.0.0" % "optional",
   "com.rojoma"        %% "rojoma-json-v3-grisu"  % "1.0.0" % "optional",
+  "org.scalatra"      %% "scalatra"            % "2.2.2" % "optional",
```
Comment (velvia) [inline build.sbt]: Note:  we really need to change `optional` to provided, for all the reps.  `optional` will result in Scalatra being pulled into every sbt assembly for every project that pulls in thirdparty-utils, which is not what we want.

Robert has some disagreements about this, but this is just how sbt-assembly currently works.   I would say it's not worth the time to fix it in SBT / assembly, and just give up ideology and agree on being practical.

File: core/src/main/scala/com/socrata/thirdparty/scalatra/RequestLogging.scala [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package com.socrata.thirdparty.scalatra
+
+import org.scalatra.ScalatraServlet
+import org.slf4j.LoggerFactory
+
+trait RequestLogging extends ScalatraServlet {
+  val logger = LoggerFactory.getLogger(getClass)
+  before() {
+    logger.info(request.getMethod + " - " + request.getRequestURI + " ? " + request.getQueryString)
```
Comment (velvia) [inline core/src/main/scala/com/socrata/thirdparty/scalatra/RequestLogging.scala]: We should also consider renaming the serving thread with some of the above info, if that's not already done.

File: build.sbt (hunk 1 above)
Comment (rjmac) [inline build.sbt]: Yeah, reality trumps philosophy every time, I'm afraid.  "provided"ify at will.

**Comment by velvia** [comment]
Reviewed to 36073c7.

**Comment by urmilan** [comment]
Addressed most of the comments in 0f97aa1.

File: core/src/main/scala/com/socrata/thirdparty/scalatra/RequestLogging.scala (hunk 2 above)
Comment (urmilan) [inline core/src/main/scala/com/socrata/thirdparty/scalatra/RequestLogging.scala]: It's not. What convention do we want to use to name threads? I can do this in a separate PR.

**Comment by velvia** [comment]
:+1: LGTM

**Pull request merged**